crates/bevy_openxr/src/interaction.rs

```diff
@@ -0,0 +1,493 @@
+            Self::FloatInput(action) => action,
+            _ => panic!(),
+        }
+    }
+    fn unwrap_vector2(self) -> xr::Action<xr::Vector2f> {
+        match self {
+            Self::Vector2Input(action) => action,
+            _ => panic!(),
+        }
+    }
+    fn unwrap_pose(self) -> xr::Action<xr::Posef> {
+        match self {
+            Self::PoseInput(action) => action,
+            _ => panic!(),
+        }
+    }
+    fn unwrap_vibration(self) -> xr::Action<xr::Haptic> {
+        match self {
+            Self::VibrationOutput(action) => action,
+            _ => panic!(),
+        }
+    }
+}
+
+pub(crate) fn pose_bindings() -> Vec<OpenXrBindingDesc> {
+    fn binding(name: &'static str, path: &'static str) -> OpenXrBindingDesc {
+        OpenXrBindingDesc {
+            name,
+            paths: vec![OpenXrActionPath {
+                profile: KHR_PROFILE,
+                path,
+            }],
+            action_type: OpenXrActionType::PoseInput,
+        }
+    }
+
+    vec![
+        binding(LEFT_GRIP_POSE, "/user/hand/left/input/grip/pose"),
+        binding(RIGHT_GRIP_POSE, "/user/hand/right/input/grip/pose"),
+        binding(LEFT_AIM_POSE, "/user/hand/left/input/aim/pose"),
+        binding(RIGHT_AIM_POSE, "/user/hand/right/input/aim/pose"),
+    ]
+}
+
+pub(crate) fn create_actions(
+    instance: &xr::Instance,
+    action_set: &xr::ActionSet,
+    bindings: &[OpenXrBindingDesc],
+) -> HashMap<&'static str, XrAction> {
+    // filter out duplicates, retain the last occurrence
+    let mut filtered_bindings = HashMap::new();
+    for binding in bindings {
+        filtered_bindings.insert(binding.name, binding);
+    }
+
+    let mut actions = HashMap::new();
+    let mut bindings_by_profile = HashMap::<&'static str, Vec<_>>::new();
+    for binding in filtered_bindings.values() {
+        let action = match binding.action_type {
+            OpenXrActionType::BinaryInput => XrAction::BinaryInput(
+                action_set
+                    .create_action(binding.name, binding.name, &[])
+                    .unwrap(),
+            ),
+            OpenXrActionType::FloatInput => XrAction::FloatInput(
+                action_set
+                    .create_action(binding.name, binding.name, &[])
+                    .unwrap(),
+            ),
+            OpenXrActionType::Vector2Input => XrAction::Vector2Input(
+                action_set
+                    .create_action(binding.name, binding.name, &[])
+                    .unwrap(),
+            ),
+            OpenXrActionType::PoseInput => XrAction::PoseInput(
+                action_set
+                    .create_action(binding.name, binding.name, &[])
+                    .unwrap(),
+            ),
+            OpenXrActionType::VibrationOutput => XrAction::VibrationOutput(
+                action_set
+                    .create_action(binding.name, binding.name, &[])
+                    .unwrap(),
+            ),
+        };
```

Comment:
You can simplify this via
```rust
let action = action_set.create_action(binding.name, binding.name, &[]).unwrap();
let action = match binding.action_type {
    OpenXrActionType::BinaryInput => XrAction::BinaryInput(action),
    // ... the rest
}
```

Reply:
This cannot be done because the actions created by `action_set.create_action()` have each one their own type argument. This is the whole point of having `XrAction` enum.

Reply:
Well in the code snippet above all of the calls to `action_set.creat_action()` all have the save arguments regardless of action type?

Reply:
Action is not just one type. The signature is `create_action<T>(...) -> Action<T>`

Reply:
Ahh gotcha
Feel free to resolve this thread ;)